emu_mps/mps_backend_impl.py

```diff
@@ -33,6 +37,56 @@
 from enum import Enum, auto
 
 
+class Statistics(Observable):
+    def __init__(
+        self,
+        evaluation_times: typing.Sequence[float] | None,
+        data: list[float],
```

Comment:
I don't like this `data`, it's very hacky
the definition of `apply` abstractmethod specifically has a `kwargs` so that you don't have to do this. Instead: retrieve for instance a `timestep_duration: float` argument from `kwargs` in `apply`

Reply:
oh sorry indeed my mistake. I looked at some subclass `apply` which has kwargs but it's just for shortcutting.
yeah reusing is better, and there is no way to achieve what I said with the pulser code